### PR TITLE
docs: Add documentation for live migration

### PR DESCRIPTION
**Pull request by likebreath**
This documentation is based on notes shared by @rbradford in https://github.com/cloud-hypervisor/cloud-hypervisor/issues/2458. These instructions have been tested on both my dev machine and Azure VM.

Signed-off-by: Bo Chen <chen.bo@intel.com>

### CODE REVIEW

File: docs/live_migration.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
+# Live Migration
+
+This document gives two examples of how to use the live migration
+support in Cloud Hypervisor: 1) local migration - migrating between two
```
Comment (sboeuf) [inline docs/live_migration.md:4]: I don't think the syntax `1)` and `2)` is rendering as you expected.

File: docs/live_migration.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    --kernel ~/workloads/vmlinux \
+    --pmem file=~/workloads/focal.raw,discard_writes=on \
+    --disk path=tmp.img \
+    --cpus boot=1 --memory size=1G \
+    --cmdline "root=/dev/pmem0p1 console=ttyS0"  \
+    --serial tty --console off --api-socket=/tmp/api1
+```
+
+Run the guest workload below (on the guest OS of the source VM), which
+features intensive virtio-blk operations. Now the console of the source
```
Comment (sboeuf) [inline docs/live_migration.md:23]: s/features/performs/

File: docs/live_migration.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+```
+
+Launch the nested destination VM (inside the guest OS of the VM 2):
+```bash
+vm-2:~$ sudo ./cloud-hypervisor --api-socket=/tmp/api2
+vm-2:~$ # setup the guest network with the following commands if needed
+vm-2:~$ sudo ip addr add 192.168.102.2/24 dev ens3
+vm-2:~$ sudo ip link set up dev ens3
+vm-2:~$ sudo ip r add default via 192.168.102.1
+vm-2:~$ ping 192.168.101.2 # This should secceed
```
Comment (sboeuf) [inline docs/live_migration.md:139]: s/secceed/succeed/

File: docs/live_migration.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+```
+> Note: If the above ping failed, please check the iptable rules on the
```
Comment (sboeuf) [inline docs/live_migration.md:141]: s/iptable/iptables/

File: docs/live_migration.md [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+between two nested VMs whose host VMs are running on the same machine.
+
+## Local Migration
+Launch the source VM with an additional disk device for testing purpose
+(on the host machine):
+```bash
+$ head -c 1M < /dev/urandom > tmp.img # create a dummy image for testing
+$ target/release/cloud-hypervisor
+    --kernel ~/workloads/vmlinux \
+    --pmem file=~/workloads/focal.raw,discard_writes=on \
```
Comment (rbradford) [inline docs/live_migration.md:15]: This example seems to be about testing the virtio-block heavy use. I think you should probably document an approach that is not trying to exercise the system.

Comment (rbradford) [inline docs/live_migration.md:15]: This use of pmem is quite niche.

File: docs/live_migration.md [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+exposing a guest image for nested source VM:
+```bash
+$ head -c 1M < /dev/urandom > tmp.img # create a dummy image for testing
+$ sudo /target/release/cloud-hypervisor \
+        --serial tty --console off \
+        --cpus boot=1 --memory size=512M \
+        --kernel vmlinux \
+        --cmdline "root=/dev/pmem0p1 console=ttyS0"  \
+        --disk path=focal-nested.raw path=tmp.img\
+        --pmem file=focal-1.raw,discard_writes=on \
```
Comment (rbradford) [inline docs/live_migration.md:76]: Again, could we just use vitio-block for both disks.

File: docs/live_migration.md (hunk 5 above)
Comment (likebreath) [reply docs/live_migration.md]: Sounds good. I removed the testing workload of virtio-block part to keep the 'local migration' example simple. 

Please let me know if you want me to simplify the  'nested-vm migration' example in the same way.

**Review by sboeuf — APPROVED** [review]

**Review by rbradford — APPROVED** [review]
@likebreath I think you've written very good documentation here for an audience of developers on how to test it. I think we also need documentation for an audience of administrators trying to deploy live migration. It would look very similar but obviously not using nested and it would also need to include details of what works and doesn't work with live migration.

**Pull request merged**